Intersect.Server/Entities/Player.cs

```diff
@@ -1670,7 +1774,235 @@ public override void Warp(Guid newMapId, float newX, float newY, bool adminWarp
             }
         }
 
-        private void MapInstanceChanged(MapController newMap)
+        /// <summary>
+        /// Warps the player on login, taking care of instance management depending on the instance type the player
+        /// is attempting to login to.
+        /// </summary>
+        public void LoginWarp()
+        {
+            if (MapId == null || MapId == Guid.Empty)
+            {
+                WarpToSpawn();
+            }
+            else
+            {
+                if (!CanChangeToInstanceType(InstanceType, true, MapId))
+                {
+                    WarpToLastOverworldLocation(true);
+                }
+                else
+                {
+                    // Will warp to spawn if we fail to create an instance for the relevant map
+                    Warp(
+                        MapId, (byte)X, (byte)Y, (byte)Dir, false, (byte)Z, false, false, InstanceType, true
```

Comment:
If any of those `false`s are default values, they should be omitted and `parameterName: <Value>` notation should be used for the remainder of the properties

Reply:
Done